dfa/preprocess.py

```diff
@@ -0,0 +1,72 @@
+import argparse
+from multiprocessing import cpu_count
+from multiprocessing.pool import Pool
+from pathlib import Path
+from typing import Dict, Union
+
+import torch
+import tqdm
+
+from dfa.audio import Audio
+from dfa.paths import Paths
+from dfa.text import Tokenizer
+from dfa.utils import get_files, read_config, pickle_binary, read_metafile
+
+
+class Preprocessor:
+    """Performs mel extraction and tokenization and stores the resulting torch tensors."""
+
+    def __init__(self, audio: Audio, tokenizer: Tokenizer,
+                 paths: Paths, text_dict: Dict[str, str]) -> None:
+        self.audio = audio
+        self.paths = paths
+        self.tokenizer = tokenizer
+        self.text_dict = text_dict
+
+    def __call__(self, wav_path: Path) -> Dict[str, Union[str, int]]:
+        item_id = wav_path.stem
+        wav = self.audio.load_wav(wav_path)
+        mel = self.audio.wav_to_mel(wav)
+        mel = torch.tensor(mel).float()
+        text = self.text_dict[item_id]
+        tokens = self.tokenizer(text)
+        tokens = torch.tensor(tokens).long()
+        torch.save(mel, self.paths.mel_dir / f'{item_id}.pt')
+        torch.save(tokens, self.paths.token_dir / f'{item_id}.pt')
```

Comment:
Root of the comment above, numpy object would be  easier to inspect from outside, unless torch objects are more efficient in some way

Reply:
Yeah, some overhead with loading torch from numpy